### DISCUSSION TITLE
How to ignore parts of depth image?

### DISCUSSION THREAD

**Discussion by thundercarrot** [lllyasviel/ControlNet#200] [Q&A]
I'm supplying my own depth image. I also have a mask that indicates where depth image pixel values are invalid.  Setting invalid depth image pixels to 0 seems wrong, as that would mean "very far away". 

How can I use this mask with the depth image model?

**Comment by lllyasviel** [comment]
Midas is inverse relative depth and the formulation is [here](https://github.com/isl-org/MiDaS/issues/52) so basically 0 means far away.
I am assuming we are talking about depth map obtained by sensors and it has some holes in it, and we will need some depth map inpainting preprocessing. It seems that people like to [use super pixels](https://www.google.com/search?q=depth+map+inpainting) to inpaint depth maps while I think iterative [joint Bilateral filter](https://en.wikipedia.org/wiki/Bilateral_filter) in the holes should be good enough
